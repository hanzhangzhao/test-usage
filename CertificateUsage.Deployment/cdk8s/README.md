# Deployment via cdk8s

Synthesize Kubernetes manifests by running:

```sh
DOCKER_LABEL=latest NODE_ENV=development npm run build
```

Authenticate with the AWS account, then apply the manifest files by running:

```sh
kubectl kustomize > applyMe.yaml
kubectl apply -f applyMe.yaml
```

## Trouble Pulling image from GitLab

If running from an empty namespace, you may see an ImagePull error and ImagePullBackOff. This means we are missing the secret in k8s for gitlab, get a deploy token and create the the secret with the following command

```sh
kubectl create secret docker-registry regcred --docker-server=registry.gitlab.com --docker-username=$USERNAME --docker-password=$PASSWORD
```

If you encounter the ImagePull and/or ImagePullBackOff errors after the first time deploying this app to a new environment or a recreated EKS cluster, you will have to do the following:

Check if the secret exists in the namespace:
```sh
kubectl get secret/regcred -n product-service
```

If not, create the secret in the namespace:

```sh
kubectl create secret docker-registry regcred --docker-server=registry.gitlab.com --docker-username=$USERNAME --docker-password=$PASSWORD --namespace=certificate-usage
```

[see more information here](https://juju.is/tutorials/using-gitlab-as-a-container-registry#7-pull-our-container)

## Environment Variables

| Name          | Description                                               | Default         |
| ------------- | --------------------------------------------------------- | --------------- |
| K8S_NAMESPACE | Namespace that all of the resources will be deployed into | certificate-usage |
| DOCKER_LABEL  | Label for the Product service that should be deployed     | latest          |