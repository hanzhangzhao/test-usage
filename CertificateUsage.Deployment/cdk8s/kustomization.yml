# DO NOT CHANGE THIS FILE
# This file is only here to include the cdk8s built deployment manifest when
# applying kustomize to the cluster.  Make changes to kustomizable in the files
# below the ../kustomize folder.
resources:
  - dist/manifest-template.k8s.yaml
commonLabels:
  slackChannel: 'project-erl-overhaul'
  team: 'brokerage'
  appName: 'certificate-usage'
  domain: 'brokerage'
