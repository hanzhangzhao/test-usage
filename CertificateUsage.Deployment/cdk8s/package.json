{
  "name": "certificate-usage.deployment",
  "version": "1.0.0",
  "main": "main.js",
  "types": "main.ts",
  "license": "Apache-2.0",
  "private": true,
  "scripts": {
    "import": "cdk8s import",
    "synth": "cdk8s synth",
    "compile": "tsc",
    "watch": "tsc -w",
    "test": "jest",
    "build": "npm run compile && npm run synth",
    "upgrade": "npm i cdk8s@latest cdk8s-cli@latest",
    "upgrade:next": "npm i cdk8s@next cdk8s-cli@next"
  },
  "dependencies": {
    "cdk8s": "^2.7.100",
    "config": "3.3.6"
  },
  "devDependencies": {
    "@types/config": "^3.3.0",
    "@types/jest": "^27.5.2",
    "@types/node": "^16.0.1",
    "cdk8s-cli": "^2.2.103",
    "jest": "^27.5.1",
    "ts-jest": "^27.1.5",
    "typescript": "^4.9.5"
  }
}
