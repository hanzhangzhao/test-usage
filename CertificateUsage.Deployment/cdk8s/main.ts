import { App, Chart } from 'cdk8s';
import * as config from 'config';
import { Construct } from 'constructs';
import {
  IntOrString,
  KubeJob,
  KubeNamespace,
  KubeServiceAccount,
  KubeSecret,
  IoK8SApiCoreV1ContainerImagePullPolicy,
  IoK8SApiCoreV1HttpGetActionScheme,
  IoK8SApiCoreV1PodSpecRestartPolicy,
  Quantity,
} from './imports/k8s';
import { WebService } from './imports/WebService';
import { StandaloneDeployment } from './imports/StandaloneDeployment';

const namespace = 'certificate-usage';
const registry = config.get<string>('docker.registry');
const listenerImageLabel = config.get<string>('listener.imageLabel')
const apiImageLabel = config.get<string>('api.imageLabel')
const migrationImageLabel = config.get<string>('migration.imageLabel');
const projectionsImageLabel = config.get<string>('projections.imageLabel');
const apmSecretToken = config.get<string>('apm.secretToken');
const migrationImageTag = `${registry}/certificate-usage/migrations:${migrationImageLabel}`;

const dbEnvs = [
  { name: 'DB__HOST', valueFrom: { secretKeyRef: { name: 'app-secrets', key: 'endpoint' } } },
  { name: 'DB__USERNAME', valueFrom: { secretKeyRef: { name: 'app-secrets', key: 'username' } } },
  { name: 'DB__PASSWORD', valueFrom: { secretKeyRef: { name: 'app-secrets', key: 'password' } } },
  { name: 'DB__PORT', valueFrom: { secretKeyRef: { name: 'app-secrets', key: 'port' } } },
  {
    name: 'DB__CONNECTIONSTRING',
    value: 'User ID=$(DB__USERNAME);Password=$(DB__PASSWORD);Host=$(DB__HOST);Port=$(DB__PORT);Search Path=certificate_usage',
  },
];

export class Deployment extends Chart {
  private registryToken = process.env.GITLAB_K8S_REG_TOKEN;
  private imagePullSecretName = 'regcred';

  private envFrom = [{ configMapRef: { name: 'app-config' } }];

  constructor(scope: Construct, id: string) {
    super(scope, id, { namespace });

    new KubeNamespace(this, 'namespace', {
      metadata: {
        name: namespace,
        labels: {
          'istio-injection': 'disabled',
        },
      },
    });

    new KubeServiceAccount(this, 'serviceaccount', {
      metadata: {
        name: 'serviceaccount',
        namespace: namespace,
      },
    });

    new KubeSecret(this, 'secret', {
      metadata: {
        name: this.imagePullSecretName,
        namespace: namespace,
      },
      type: 'kubernetes.io/dockerconfigjson',
      stringData: {
        '.dockerconfigjson': `{"auths":{"registry.gitlab.com":{"auth":"${Buffer.from(
          `gitlab-ci:${this.registryToken}`
        ).toString('base64')}"}}}`,
      },
    });

    new StandaloneDeployment(this, 'certificate-usage-member-listener', {
      namespace,
      name: 'certificate-usage-member-listener',
      image: `${registry}/certificate-usage:${listenerImageLabel}`,
      imagePullSecretName: this.imagePullSecretName,
      envFrom: this.envFrom,
      env: [
        ...dbEnvs,
        { name: 'ELASTICAPM__SECRETTOKEN', value: apmSecretToken },
        { name: 'ASPNETCORE_URLS', value: 'http://0.0.0.0:8080' }
      ],
      livenessProbe: {
        exec: {
          command: [
            "/bin/sh",
            "-c",
            "./health.sh"
          ]
        },
        initialDelaySeconds: 10,
        periodSeconds: 5,
        timeoutSeconds: 5,
      },
      readinessProbe: {
        exec: {
          command: [
            "/bin/sh",
            "-c",
            "./health.sh"
          ]
        },
        failureThreshold: 6,
        initialDelaySeconds: 10,
        periodSeconds: 5,
        timeoutSeconds: 5,
      },
      initContainers: [{
        name: 'projections-init',
        image: `${registry}/certificate-usage/projections:${projectionsImageLabel}`,
        envFrom: this.envFrom,
        imagePullPolicy: IoK8SApiCoreV1ContainerImagePullPolicy.ALWAYS,
        securityContext: { allowPrivilegeEscalation: false },
      }],
    });

    new WebService(this, 'certificate-usage', {
      namespace,
      name: 'certificate-usage',
      image: `${registry}/certificate-usage/api:${apiImageLabel}`,
      imagePullSecretName: this.imagePullSecretName,
      minReplicas: 2,
      portName: 'http',
      port: 8080,
      containerPort: 8080,
      env: [
        ...dbEnvs,
        { name: 'ELASTICAPM__SECRETTOKEN', value: apmSecretToken },
        { name: 'ASPNETCORE_URLS', value: 'http://0.0.0.0:8080' }
      ],
      envFrom: this.envFrom,
      canaryAnalysis: {
        interval: '30s',
        iterations: 5,
        threshold: 4,
      },
      livenessProbe: {
        httpGet: {
          httpHeaders: [{ name: 'User-Agent', value: 'k8s liveness probe' }],
          port: IntOrString.fromNumber(8080),
          path: '/health',
          scheme: IoK8SApiCoreV1HttpGetActionScheme.HTTP,
        },
        initialDelaySeconds: 10,
        periodSeconds: 5,
        timeoutSeconds: 5,
      },
      readinessProbe: {
        httpGet: {
          httpHeaders: [{ name: 'User-Agent', value: 'k8s readiness probe' }],
          port: IntOrString.fromNumber(8080),
          path: '/health',
          scheme: IoK8SApiCoreV1HttpGetActionScheme.HTTP,
        },
        failureThreshold: 6,
        initialDelaySeconds: 10,
        periodSeconds: 5,
        timeoutSeconds: 5,
      },
    });

    this.createMigrationJob();
  }

  createMigrationJob() {
    new KubeJob(this, 'certificate-usage-db-migration', {
      metadata: {
        name: 'certificate-usage-db-migration',
        namespace: namespace
      },
      spec: {
        ttlSecondsAfterFinished: 100,
        template: {
          spec: {
            imagePullSecrets: [{ name: 'regcred' }],
            restartPolicy: IoK8SApiCoreV1PodSpecRestartPolicy.NEVER,
            containers: [
              {
                name: 'certificate-usage-db-migration',
                image: migrationImageTag,
                env: [
                  ...dbEnvs,
                  { name: 'ELASTICAPM__SECRETTOKEN', value: apmSecretToken },
                ],
                resources: {
                  requests: {
                    cpu: Quantity.fromString('100m'),
                    memory: Quantity.fromString('128Mi')
                  },
                  limits: {
                    cpu: Quantity.fromString('200m'),
                    memory: Quantity.fromString('256Mi')
                  }
                },
                securityContext: { allowPrivilegeEscalation: false }
              }
            ]
          }
        }
      }
    })
  }
}

const app = new App();
new Deployment(app, 'manifest-template');
app.synth();
