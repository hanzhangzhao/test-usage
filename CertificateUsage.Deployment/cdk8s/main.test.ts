import {Deployment} from './main';
import {Testing} from 'cdk8s';

describe('Kubernetes Deployment', () => {
  test('Kubernetes manifest', () => {
    const app = Testing.app();
    const chart = new Deployment(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
