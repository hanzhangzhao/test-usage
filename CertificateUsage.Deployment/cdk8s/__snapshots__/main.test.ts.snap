// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kubernetes Deployment Kubernetes manifest 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "istio-injection": "disabled",
      },
      "name": "certificate-usage",
      "namespace": "certificate-usage",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "name": "serviceaccount",
      "namespace": "certificate-usage",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Secret",
    "metadata": Object {
      "name": "regcred",
      "namespace": "certificate-usage",
    },
    "stringData": Object {
      ".dockerconfigjson": "{\\"auths\\":{\\"registry.gitlab.com\\":{\\"auth\\":\\"Z2l0bGFiLWNpOnVuZGVmaW5lZA==\\"}}}",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "test-chart-certificate-usage-member-listener-c8ec7b87",
      },
      "name": "certificate-usage-member-listener",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "selector": Object {
        "matchLabels": Object {
          "app": "test-chart-certificate-usage-member-listener-c8ec7b87",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-chart-certificate-usage-member-listener-c8ec7b87",
          },
          "name": "certificate-usage-member-listener",
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [],
              "env": Array [
                Object {
                  "name": "DB__HOST",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "endpoint",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__USERNAME",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "username",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PASSWORD",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "password",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PORT",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "port",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__CONNECTIONSTRING",
                  "value": "User ID=$(DB__USERNAME);Password=$(DB__PASSWORD);Host=$(DB__HOST);Port=$(DB__PORT);",
                },
                Object {
                  "name": "ELASTICAPM__SECRETTOKEN",
                  "value": "some key",
                },
                Object {
                  "name": "ASPNETCORE_URLS",
                  "value": "http://0.0.0.0:8080",
                },
              ],
              "envFrom": Array [
                Object {
                  "configMapRef": Object {
                    "name": "app-config",
                  },
                },
              ],
              "image": "registry.gitlab.com/sterlingcapitalbrokers/certificate-usage:latest",
              "imagePullPolicy": "Always",
              "livenessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "/bin/sh",
                    "-c",
                    "./health.sh",
                  ],
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 5,
                "timeoutSeconds": 5,
              },
              "name": "certificate-usage-member-listener",
              "readinessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "/bin/sh",
                    "-c",
                    "./health.sh",
                  ],
                },
                "failureThreshold": 6,
                "initialDelaySeconds": 10,
                "periodSeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": 1,
                  "memory": "256Mi",
                },
                "requests": Object {
                  "cpu": 1,
                  "memory": "128Mi",
                },
              },
              "securityContext": Object {
                "allowPrivilegeEscalation": false,
              },
            },
          ],
          "imagePullSecrets": Array [
            Object {
              "name": "regcred",
            },
          ],
          "serviceAccountName": "serviceaccount",
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "test-chart-certificate-usage-c833da23",
      },
      "name": "certificate-usage",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "selector": Object {
        "matchLabels": Object {
          "app": "test-chart-certificate-usage-c833da23",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-chart-certificate-usage-c833da23",
          },
          "name": "certificate-usage",
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [],
              "env": Array [
                Object {
                  "name": "DB__HOST",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "endpoint",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__USERNAME",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "username",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PASSWORD",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "password",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PORT",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "port",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__CONNECTIONSTRING",
                  "value": "User ID=$(DB__USERNAME);Password=$(DB__PASSWORD);Host=$(DB__HOST);Port=$(DB__PORT);",
                },
                Object {
                  "name": "ELASTICAPM__SECRETTOKEN",
                  "value": "some key",
                },
                Object {
                  "name": "ASPNETCORE_URLS",
                  "value": "http://0.0.0.0:8080",
                },
              ],
              "envFrom": Array [
                Object {
                  "configMapRef": Object {
                    "name": "app-config",
                  },
                },
              ],
              "image": "registry.gitlab.com/sterlingcapitalbrokers/certificate-usage/api:latest",
              "imagePullPolicy": "Always",
              "livenessProbe": Object {
                "httpGet": Object {
                  "httpHeaders": Array [
                    Object {
                      "name": "User-Agent",
                      "value": "k8s liveness probe",
                    },
                  ],
                  "path": "/health",
                  "port": 8080,
                  "scheme": "HTTP",
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 5,
                "timeoutSeconds": 5,
              },
              "name": "certificate-usage",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "readinessProbe": Object {
                "failureThreshold": 6,
                "httpGet": Object {
                  "httpHeaders": Array [
                    Object {
                      "name": "User-Agent",
                      "value": "k8s readiness probe",
                    },
                  ],
                  "path": "/health",
                  "port": 8080,
                  "scheme": "HTTP",
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": 1,
                  "memory": "256Mi",
                },
                "requests": Object {
                  "cpu": 1,
                  "memory": "128Mi",
                },
              },
              "securityContext": Object {
                "allowPrivilegeEscalation": false,
              },
            },
          ],
          "imagePullSecrets": Array [
            Object {
              "name": "regcred",
            },
          ],
          "serviceAccountName": "serviceaccount",
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "certificate-usage",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "8080-8080",
          "port": 8080,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "test-chart-certificate-usage-c833da23",
      },
    },
  },
  Object {
    "apiVersion": "autoscaling/v2beta2",
    "kind": "HorizontalPodAutoscaler",
    "metadata": Object {
      "name": "certificate-usage",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "maxReplicas": 2,
      "metrics": Array [
        Object {
          "resource": Object {
            "name": "cpu",
            "target": Object {
              "averageUtilization": 50,
              "type": "Utilization",
            },
          },
          "type": "Resource",
        },
      ],
      "minReplicas": 1,
      "scaleTargetRef": Object {
        "apiVersion": "apps/v1",
        "kind": "Deployment",
        "name": "certificate-usage",
      },
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "VirtualService",
    "metadata": Object {
      "name": "test-chart-certificate-usage-vs-c8f5ca2c",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "exportTo": Array [
        "*",
      ],
      "hosts": Array [
        "certificate-usage.certificate-usage.svc.cluster.local",
      ],
      "http": Array [
        Object {
          "route": Array [
            Object {
              "destination": Object {
                "host": "certificate-usage.certificate-usage.svc.cluster.local",
              },
            },
          ],
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "DestinationRule",
    "metadata": Object {
      "name": "test-chart-certificate-usage-dr-c87beb1c",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "exportTo": Array [
        "*",
      ],
      "host": "certificate-usage.certificate-usage.svc.cluster.local",
      "trafficPolicy": Object {
        "loadBalancer": Object {
          "simple": "LEAST_CONN",
        },
      },
    },
  },
  Object {
    "apiVersion": "flagger.app/v1beta1",
    "kind": "Canary",
    "metadata": Object {
      "name": "certificate-usage",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "analysis": Object {
        "interval": "30s",
        "iterations": 5,
        "threshold": 4,
      },
      "autoscalerRef": Object {
        "apiVersion": "autoscaling/v2beta2",
        "kind": "HorizontalPodAutoscaler",
        "name": "certificate-usage",
      },
      "service": Object {
        "canary": Object {
          "annotations": Object {
            "konghq.com/override": "certificate-usage",
          },
        },
        "port": 8080,
        "portName": "http",
        "primary": Object {
          "annotations": Object {
            "konghq.com/override": "certificate-usage",
          },
        },
      },
      "targetRef": Object {
        "apiVersion": "apps/v1",
        "kind": "Deployment",
        "name": "certificate-usage",
      },
    },
  },
  Object {
    "apiVersion": "ec2.aws.crossplane.io/v1beta1",
    "kind": "SecurityGroup",
    "metadata": Object {
      "name": "certificate-usage-security-group",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "forProvider": Object {
        "description": "Enable rds access",
        "egress": Array [
          Object {
            "fromPort": 0,
            "ipProtocol": "tcp",
            "ipRanges": Array [
              Object {
                "cidrIp": "0.0.0.0/0",
              },
            ],
            "toPort": 0,
          },
        ],
        "groupName": "database-security-group",
        "ingress": Array [
          Object {
            "fromPort": 5432,
            "ipProtocol": "tcp",
            "ipRanges": Array [
              Object {
                "cidrIp": "172.31.0.0/16",
              },
            ],
            "toPort": 5432,
          },
        ],
        "region": "ca-central-1",
        "vpcId": "vpc-000468bf7dd41a01d",
      },
    },
  },
  Object {
    "apiVersion": "database.aws.crossplane.io/v1beta1",
    "kind": "DBSubnetGroup",
    "metadata": Object {
      "name": "certificate-usage-db-subnet-group",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "forProvider": Object {
        "description": "subnet group for certificate-usage",
        "region": "ca-central-1",
        "subnetIds": Array [
          "subnet-08e840f1c75bc8bcd",
          "subnet-0c88e18012e8ac7b6",
          "subnet-0064b60b74121308a",
        ],
      },
    },
  },
  Object {
    "apiVersion": "database.aws.crossplane.io/v1beta1",
    "kind": "RDSInstance",
    "metadata": Object {
      "name": "certificate-usage-db",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "forProvider": Object {
        "allocatedStorage": 10,
        "backupRetentionPeriod": 1,
        "dbInstanceClass": "db.t4g.micro",
        "dbName": "certificate_usage_service",
        "dbSubnetGroupNameRef": Object {
          "name": "certificate-usage-db-subnet-group",
        },
        "enablePerformanceInsights": false,
        "engine": "postgres",
        "engineVersion": "15.3",
        "masterUsername": "certificate_usage_service",
        "multiAZ": false,
        "region": "ca-central-1",
        "skipFinalSnapshotBeforeDeletion": true,
        "storageEncrypted": true,
        "tags": Array [
          Object {
            "key": "app",
            "value": "certificate-usage",
          },
          Object {
            "key": "namespace",
            "value": "certificate-usage",
          },
          Object {
            "key": "team",
            "value": "brokerage",
          },
        ],
        "vpcSecurityGroupIDRefs": Array [
          Object {
            "name": "certificate-usage-security-group",
          },
        ],
      },
      "writeConnectionSecretToRef": Object {
        "name": "db-cred",
        "namespace": "certificate-usage",
      },
    },
  },
  Object {
    "apiVersion": "batch/v1",
    "kind": "Job",
    "metadata": Object {
      "name": "certificate-usage-db-migration",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "template": Object {
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "DB__HOST",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "endpoint",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__USERNAME",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "username",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PASSWORD",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "password",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__PORT",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "port",
                      "name": "db-cred",
                    },
                  },
                },
                Object {
                  "name": "DB__CONNECTIONSTRING",
                  "value": "User ID=$(DB__USERNAME);Password=$(DB__PASSWORD);Host=$(DB__HOST);Port=$(DB__PORT);",
                },
                Object {
                  "name": "ELASTICAPM__SECRETTOKEN",
                  "value": "some key",
                },
              ],
              "image": "registry.gitlab.com/sterlingcapitalbrokers/certificate-usage/migrations:latest",
              "name": "certificate-usage-db-migration",
              "resources": Object {
                "limits": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
                "requests": Object {
                  "cpu": "100m",
                  "memory": "128Mi",
                },
              },
              "securityContext": Object {
                "allowPrivilegeEscalation": false,
              },
            },
          ],
          "imagePullSecrets": Array [
            Object {
              "name": "regcred",
            },
          ],
          "restartPolicy": "Never",
        },
      },
      "ttlSecondsAfterFinished": 100,
    },
  },
  Object {
    "apiVersion": "s3.aws.crossplane.io/v1beta1",
    "kind": "Bucket",
    "metadata": Object {
      "annotations": Object {
        "crossplane.io/external-name": "monthly-billing-development",
      },
      "name": "monthly-billing-bucket",
      "namespace": "certificate-usage",
    },
    "spec": Object {
      "forProvider": Object {
        "locationConstraint": "ca-central-1",
        "objectOwnership": "BucketOwnerEnforced",
        "paymentConfiguration": Object {
          "payer": "BucketOwner",
        },
        "publicAccessBlockConfiguration": Object {
          "blockPublicAcls": true,
          "blockPublicPolicy": true,
          "ignorePublicAcls": true,
          "restrictPublicBuckets": true,
        },
        "serverSideEncryptionConfiguration": Object {
          "rules": Array [
            Object {
              "applyServerSideEncryptionByDefault": Object {
                "sseAlgorithm": "AES256",
              },
            },
          ],
        },
        "versioningConfiguration": Object {
          "status": "Enabled",
        },
      },
      "writeConnectionSecretToRef": Object {
        "name": "bucket-access",
        "namespace": "certificate-usage",
      },
    },
  },
]
`;
